tests/foreman/cli/test_logging.py

```diff
@@ -0,0 +1,189 @@
+        with open("/var/tmp/logfile_from_foreman_core", "r") as logfile:
+            for line in logfile:
+                if re.search(r'Started GET \"\/api/hosts\?page=1', line):
+                    self.logger.info('Found the line with GET to hosts API')
+                    GET_line_found = 1
+                    # Confirm the request ID was logged in the line with GET
+                    match = re.search(r'\[I\|app\|\w{8}\]', line)
+                    if match:
+                        self.logger.info("Request ID found for logging from foreman core")
+                    else:
+                        assert match == 1, "Request ID not found"
+        assert GET_line_found == 1, "The GET command to list hosts was not found in logs."
+
+    def test_positive_logging_from_foreman_proxy(self):
+        """Check PUT to Smart Proxy API to refresh the features is logged and has request ID."""
+        PUT_line_found = 0
+        # log the start of the test in case of problems with slow systems and time zones
+        self.logger.info("Testing test_positive_logging_from_proxy")
+        with ssh.get_connection() as connection:
+            result = connection.run('hammer proxy refresh-features --id 1')
+            self.assertEqual(result.return_code, 0)
+            # extract last ten lines from log
+            result = connection.run('tail -20 /var/log/foreman/production.log > \
+                        /var/tmp/logfile_from_proxy')
+            self.assertEqual(result.return_code, 0, "BASH command error")
+        # use same location on remote and local for log file extract
+        logfile_location = '/var/tmp/logfile_from_proxy'
+        ssh.download_file(logfile_location)
+        # search the log file extract for the line with GET to host API
+        with open("/var/tmp/logfile_from_proxy", "r") as logfile:
+            for line in logfile:
+                if re.search(r'Started PUT \"\/api\/smart_proxies\/1\/refresh', line):
+                    self.logger.info('Found the line with PUT to foreman proxy API')
+                    PUT_line_found = 1
+                    # Confirm the request ID was logged in the line with PUT
+                    match = re.search(r'\[I\|app\|\w{8}\]', line)
+                    if match:
+                        self.logger.info("Request ID found for logging from foreman proxy")
+                    else:
+                        assert match == 1, "Request ID not found"
+        assert PUT_line_found == 1, "The PUT command to refresh proxies was not found in logs."
+
+    def test_positive_logging_from_candlepin(self):
+        """Check logging after load or manifest refresh."""
+        GET_line_found = 0
+        POST_line_found = 0
+        with ssh.get_connection() as connection:
+            # get the number of lines in the log before the test
+            line_count_1 = connection.run('wc -l < \
+                                    /var/log/foreman/production.log'.format(),
```

Comment:
This path is used three times in this test, which means you should think about putting it in variable.

Reply:
I have updated all tests to use variables for source and test log files